src/Utils/request/useInfiniteQuery.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useState } from "react";
+
+import { RESULTS_PER_PAGE_LIMIT } from "@/common/constants";
+
+import { PaginatedResponse, QueryRoute } from "@/Utils/request/types";
+import useQuery, { QueryOptions } from "@/Utils/request/useQuery";
+
+export interface InfiniteQueryOptions<TItem>
+  extends QueryOptions<PaginatedResponse<TItem>> {
+  deduplicateBy?: (item: TItem) => string | number;
+}
+
+export function useInfiniteQuery<TItem>(
+  route: QueryRoute<PaginatedResponse<TItem>>,
+  options?: InfiniteQueryOptions<TItem>,
+) {
+  const [items, setItems] = useState<TItem[]>([]);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [totalCount, setTotalCount] = useState<number>(1);
```

Comment:
the initial state of this would be undefined right? we do not know if its 1 initially right?

```suggestion
  const [totalCount, setTotalCount] = useState<number>();
```